compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -614,6 +619,32 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: LocalDefId) {
     }
 }
 
+fn suggest_wrapping_unsafe_block(
+    tcx: TyCtxt<'_>,
+    def_id: LocalDefId,
+    err: &mut DiagnosticBuilder<'_, ()>,
+) {
+    let body_span = tcx
+        .hir()
+        .body(tcx.hir().body_owned_by(tcx.hir().local_def_id_to_hir_id(def_id)))
+        .value
+        .span;
+
+    let suggestion = vec![
+        (
+            tcx.sess.source_map().start_point(body_span).shrink_to_hi(),
+            " /* FIXME */ unsafe {".into(),
+        ),
+        (tcx.sess.source_map().end_point(body_span).shrink_to_lo(), "}".into()),
+    ];
+
+    err.multipart_suggestion_verbose(
+        "consider wrapping the function in an unsafe block",
+        suggestion,
+        Applicability::MaybeIncorrect,
```

Comment:
perhaps HasPlaceholders?

Reply:
The docs for `Applicability` state
> The suggestion cannot be applied automatically because it will not result in valid Rust code.

But I think the goal is to be able to apply a "quick fix" to make the code compile. Can `HasPlaceholders` suggestion still be applied "auto-manually"? I have never used rustfix (:sweat_smile:) and I'm not sure how the different `Applicability` settings behave for the end user.

Reply:
Maybe not, good point.